test/test/client_agent/ClientAgentCan.cpp

```diff
@@ -0,0 +1,15 @@
+#include <gtest/gtest.h>
+#include <Client.hpp>
+#include <thread>
+
+#include "ClientAgentCan.hpp"
+
+TEST_P(ClientAgentCan, PingFromClientToAgent)
+{
+    ASSERT_NO_FATAL_FAILURE(client_can_.ping_agent());
+}
+
+INSTANTIATE_TEST_CASE_P(
+    CanTransports,
+    ClientAgentCan,
+    ::testing::Values(MiddlewareKind::FASTDDS, MiddlewareKind::FASTRTPS, MiddlewareKind::CED));
```

Comment:
Maybe if we are testing just a CAN transport is not required to run it over all the middlewares. Thoughts?

Reply:
```suggestion
    ::testing::Values(MiddlewareKind::FASTDDS));
```
Fine by me